language/java/develop.md

```diff
@@ -0,0 +1,198 @@
+
+## Introduction
+
+In this module, we’ll walk through setting up a local development environment for the application we built in the previous modules. We’ll use Docker to build our images and Docker Compose to make everything a whole lot easier.
+
+## Run a database in a container
+
+First, we’ll take a look at running a database in a container and how we use volumes and networking to persist our data and allow our application to talk with the database. Then we’ll pull everything together into a Compose file which allows us to setup and run a local development environment with one command. Finally, we’ll take a look at connecting a debugger to our application running inside a container.
+
+Instead of downloading MySQL, installing, configuring, and then running the MySQL database as a service, we can use the Docker Official Image for MySQL and run it in a container.
+
+Before we run MySQL in a container, we'll create a couple of volumes that Docker can manage to store our persistent data and configuration. Let’s use the managed volumes feature that Docker provides instead of using bind mounts. You can read all about [Using volumes](../../storage/volumes.md) in our documentation.
+
+Let’s create our volumes now. We’ll create one for the data and one for configuration of MySQL.
+
+```console
+$ docker volume create mysql
+$ docker volume create mysql_config
+```
+
+Now we’ll create a network that our application and database will use to talk to each other. The network is called a user-defined bridge network and gives us a nice DNS lookup service which we can use when creating our connection string.
+
+```console
+$ docker network create mysqlnet
+```
+
+Now we can run MySQL in a container and attach to the volumes and network we created above. Docker pulls the image from Hub and runs it for you locally.
+
+```console
+$ docker run -it --rm -d -v mysql_data:/var/lib/mysql \
+-v mysql_config:/etc/mysql/conf.d \
+--network mysql \
+--name mysqlserver \
+-e MYSQL_USER=petclinic -e MYSQL_PASSWORD=petclinic \
+-e MYSQL_ROOT_PASSWORD=root -e MYSQL_DATABASE=petclinic \
+-p 3306:3306 mysql:8.0.23
+```
+
+Okay, now that we have a running MySQL, let’s update our Dockerfile to activate the MySQL Spring profile defined in the application and switch from an in-memory H2 database to the MySQL server we just created.
+
+We only need to add the MySQL profil as an argument to the `CMD` definition.
+
+```console
+CMD ["./mvnw", "spring-boot:run", "-Dspring-boot.run.profiles=mysql"]
+```
+
+Let's build our image
+
+```console
+$ docker build --tag java-docker .
+```
+
+Now, let’s run our container. This time, we need to set the `MYSQL_URL` environment variable so that our application knows what connection string to use to access the database. We’ll do this using the `docker run` command.
+
+```console
+$ docker run -rm -d \
+--name springboot-server \
```

Comment:
And there's another problem in language/java/run-containers.md  line 167 we already started a container with that name. Should we prepend a step to remove the container?

Reply:
Discussed with @StefanScherer and agreed to leave this 'as is' and update it later based on user feedback.